src/components/ComponentSettingsModal.jsx

```diff
@@ -227,7 +227,7 @@ const ComponentSettingsModal = ({
                   />
                 </div>
                 <SaveDeleteButtons 
-                  isDisabled={isNewFile ? hasErrors : (hasErrors || !sha)}
+                  isDisabled={isNewFile && !isPost ? (hasErrors || !fileUrl) : isNewFile && isPost ? hasErrors : (hasErrors || !sha)}
```

Comment:
For resources, we should perform this check regardless of whether it is a new file, i.e.

`isDisabled={!isPost ? (hasErrors || !fileUrl) : isNewFile && isPost ? hasErrors : (hasErrors || !sha)}`

Reply:
Fixed in acbe42c, `setHasErrors` contains logic to differentiate between errors for resource posts and resources files